.circleci/config.yml

```diff
@@ -1,14 +1,49 @@
-# Golang CircleCI 2.0 configuration file
-#
-# Check https://circleci.com/docs/2.0/language-go/ for more details
+defaults: &defaults
+  working_directory: /go/src/github.com/ligato/networkservicemesh
+  docker:
+    - image: circleci/golang:1.10
+
 version: 2
 jobs:
+  sanity-check:
+    <<: *defaults
+    steps:
+      - checkout
+      - run: sudo apt-get install yamllint shellcheck python3-pkg-resources
+      - run: yamllint -c .yamllint.yml $(git ls-files '*.yaml' '*.yml' | grep -v 'vendor/')
+      - run: make check dep-check
+
   build:
-    docker: # use the docker executor type; machine and macos executors are also supported
-      - image: circleci/golang:1.9
+    <<: *defaults
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run: echo "export COMMIT=${CIRCLE_SHA1::8}" >> $BASH_ENV
+      - run: echo "COMMIT=$COMMIT"
+      - run: go install ./vendor/k8s.io/kube-openapi/cmd/openapi-gen
+      - run: make verify docker-build docker-push
```

Comment:
@platoff Lets not do the docker push from here just yet, first lets get the build working, and then lets decide when we want to shift docker push to here instead of the travis build. Otherwise we'll be pushing twice, but also lets figure out if we want to do docker push from this build or from travis-ci.

Reply:
The last step I need to implement is to make freshly built images available on deployed k8s cluster. I want to make them available via docker hub, e.g. push freshly built images and let k8s pull them for integration tests. There should not be conflicts with Travis. Travis push images tagged as below:

```
if [ "${TRAVIS_PULL_REQUEST}" == "false" ]; then export TAG=`if [ "${TRAVIS_BRANCH}" == "master" ]; then echo "latest"; else echo ${TRAVIS_BRANCH}; fi` && make docker-push ; fi
```

I can provide CircleCI's generated tags which would not affect existing infrastructure and do not break anything. Thoughts?

Reply:
@mestery Obviously, there are other ways to make freshly build images available on cluster deployed in the cloud, but I believe this way is most portable (consider deployment to GKE, etc where we can't potentially ssh into, or may have problems build docker images and/or make them available to a node). As for tagging, Circle CI will not use `latest` as well as we can use bytes 8-16 of commit sha not to interfere with travis, which use bytes 0-8

Reply:
My thoughts are: on the plus side, pushing the images to somewhere like docker hub with a CircleCI generated tag would allow us to pull them from not only Packet, but also from other cloud provides like AWS/GCE/Azure etc as we get them up and online.

@mestery What are your thoughts on @platoff 's idea?

Reply:
Guys, my point is Travis-ci is already doing the push, and we need to coordinate this while we transition. The other thought is we have the packet.net stuff use the Travis artifacts for now. But clearly coordination and how we want to use both packet.net and Travis is needed before this is merged.